packages/filter/src/components/tag/TagCondition.tsx

```diff
@@ -16,14 +16,23 @@ import { Remove } from '../icons';
 
 
 export const TagCondition: React.FC<ITagConditionProps> = ({
+  tagConditionsRef,
   currentFilterState,
   filterSetting,
   validationError,
+  filterTheme,
   onChangeFilter,
   onRemoveFilter,
 }) => {
   const { key: guid, condition, column } = currentFilterState;
 
+  const tagConditionSelectStyles = {
+    menuPortal: (css) => ({
+      ...css,
+      zIndex: filterTheme.tagConditionSelectZIndex,
+    }),
+  };
```

Comment:
I think it is redundant (same for another comment). It is necessary to add useMemo in parts of code with a high computational load. In other cases it can be optional.

I like this article about this topic: https://kentcdodds.com/blog/usememo-and-usecallback and I absolutely agree with the author.

Reply:
po moemu statya hernya)

Reply:
This article says that in most cases, everything works well without caching. But this component is very complex (judging by the code in the repository) and there are doubts, but are you sure that this will not lead to unnecessary calculations? How did you know that?